src/Parameters.jl

```diff
@@ -253,6 +253,12 @@ covariance_transform_diagonal(Π::ScaledLogitNormal, X) = - (Π.upper_bound - Π
 ##### Free parameters
 #####
 
+"""
+    struct FreeParameters{N, P}
+
+A container for free parameters that includes the parameter names and their
+corresponding prior distributions.
+"""
```

Comment:
I think we shouldn't document the structs cause then they show up in help (but they aren't user facing...)